spec/rubocop/cop/betterment/use_global_strict_loading/for_associations_spec.rb

```diff
@@ -0,0 +1,113 @@
+  context 'when `strict_loading` is set in an association' do
+    it 'registers an offense for belongs_to' do
+      expect_offense(<<~RUBY)
+        class MyModel < ApplicationRecord
+          belongs_to :user, strict_loading: false # preserve this comment
+          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Do not set `:strict_loading` in ActiveRecord associations.
+        end
+      RUBY
+
+      expect_correction(<<~RUBY)
+        class MyModel < ApplicationRecord
+          belongs_to :user # preserve this comment
+        end
+      RUBY
+    end
+
+    it 'registers an offense for has_and_belongs_to_many' do
+      expect_offense(<<~RUBY)
+        class MyModel < ApplicationRecord
+          has_and_belongs_to_many :tags, strict_loading: true # preserve this comment
+          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Do not set `:strict_loading` in ActiveRecord associations.
+        end
+      RUBY
+
+      expect_correction(<<~RUBY)
+        class MyModel < ApplicationRecord
+          has_and_belongs_to_many :tags # preserve this comment
+        end
+      RUBY
+    end
+
+    it 'registers an offense for has_many' do
+      expect_offense(<<~RUBY)
+        class MyModel < ApplicationRecord
+          has_many :posts, strict_loading: true # preserve this comment
+          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Do not set `:strict_loading` in ActiveRecord associations.
+        end
+      RUBY
+
+      expect_correction(<<~RUBY)
+        class MyModel < ApplicationRecord
+          has_many :posts # preserve this comment
+        end
+      RUBY
+    end
+
+    it 'registers an offense for has_one' do
+      expect_offense(<<~RUBY)
+        class MyModel < ApplicationRecord
+          has_one :account, strict_loading: true # preserve this comment
+          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Do not set `:strict_loading` in ActiveRecord associations.
+        end
+      RUBY
+
+      expect_correction(<<~RUBY)
+        class MyModel < ApplicationRecord
+          has_one :account # preserve this comment
+        end
+      RUBY
+    end
+
+    it 'preserves other options' do
+      expect_offense(<<~RUBY)
+        class MyModel < ApplicationRecord
+          has_many :posts, strict_loading: true, dependent: :destroy # preserve this comment
+          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Do not set `:strict_loading` in ActiveRecord associations.
+        end
+      RUBY
+
+      expect_correction(<<~RUBY)
+        class MyModel < ApplicationRecord
+          has_many :posts, dependent: :destroy # preserve this comment
+        end
+      RUBY
+    end
+
+    it 'preserves multiline options' do
+      expect_offense(<<~RUBY)
+        class MyModel < ApplicationRecord
+          has_many :posts,        # preserve this comment
+          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Do not set `:strict_loading` in ActiveRecord associations.
```

Comment:
__comment:__ This was a bit annoying that I couldn't figure out how to make it also highlight the actual option on the next line. However, I feel it's probably "close enough" that anyone could figure out that the problem is associated with this specific `has_many` call.

For this cop to work the auto-correct is not strictly required; and I should note the autocorrect *does* work in this case, the problem is mostly wrt what parts are highlighted, which is a secondary concern.